sklearn/inspection/_partial_dependence.py

```diff
@@ -351,19 +339,25 @@ def partial_dependence(estimator, X, features, response_method='auto',
         if (isinstance(estimator, BaseGradientBoosting) and
                 estimator.init is None):
             method = 'recursion'
-        elif isinstance(estimator, BaseHistGradientBoosting):
+        elif isinstance(estimator, (BaseHistGradientBoosting,
```

Comment:
do you think that it would make sense to avoid this `isinstance` and check for `hasattr(estimator, 'compute_partial_dependence')` or a method that could be shared across all these estimators?

I could think that library as xgboost or lightgbm or tree base could expose the same method without us checking for type.

Reply:
that's a nice idea but that means we need to support a new public API, i.e. the interface of `_compute_partial_dependence_recursion` would now be fixed

We would also still need to hardcoded list of supported estimators for the error message.

Maybe we can keep that in mind for later?

Reply:
Yep we could check how other libraries are exposing this and put all these info in an issue